src/content/blog/astro-5100.mdx

```diff
@@ -0,0 +1,280 @@
+
+Most people should enable automatic styles for responsive images in your `astro.config.mjs` file:
+
+```js title="astro.config.mjs" ins={3}
+export default defineConfig({
+	image: {
+		responsiveStyles: true,
+	},
+});
+```
+
+You enable responsive images by setting a `layout`, either globally in your Astro config or on a per-image basis.
+
+```js title="astro.config.mjs" {4}
+export default defineConfig({
+	image: {
+		responsiveStyles: true,
+		layout: 'constrained',
+	},
+});
+```
+
+**Layout options:** Choose from `constrained`, `fixed`, or `full-width` layouts to control how your images behave:
+
+```astro
+<Image
+	src="/hero.jpg"
+	alt="A panoramic view of the mountains"
+	layout="full-width"
+/>
+```
+
+**Priority loading:** Use the new `priority` prop to optimize critical images that appear above the fold:
+
+```astro
+<Image src="/hero.jpg" alt="Hero image" priority />
+```
+
+When you add `priority`, Astro automatically sets `loading="eager"`, `decoding="sync"`, and `fetchpriority="high"` to ensure the image loads immediately. Use this sparingly – ideally only one image per page should have priority loading. A good rule of thumb is to check if your [LCP element](https://web.dev/lcp/#lcp-element) is an image, and if so, set the `priority` flag on it.
+
+**Enhanced cropping controls:** The `fit` and `position` properties give you precise control over how images are cropped and positioned:
+
+```astro
+<Image
+	src="/profile.jpg"
+	alt="Profile photo"
+	fit="cover"
+	position="center top"
+	width={300}
+	height={300}
+/>
+```
+
+### Upgrading from experimental responsive images
+
+If you were using experimental responsive images, you should remove the experimental flag from your `astro.config.mjs` file:
+
+```js title="astro.config.mjs" del={3} ins={6}
+export default defineConfig({
+	experimental: {
+		responsiveImages: true,
+	},
+	image: {
+		responsiveStyles: true,
+	},
+});
+```
+
+For complete details, see the [Images guide](https://docs.astro.build/en/guides/images/).
+
+## Improvements to experimental Content Security Policy
+
+Astro 5.9 shipped experimental support for Content Security Policy (CSP) meta tags, bringing powerful security features to your Astro sites. In 5.10, we've made several improvements based on community feedback. The most important is that we now support generating CSP headers, including for static sites.
+
+On-demand rendered pages will now send Response headers for CSP, rather than meta tags. This improves performance in Chrome, and adds support for more directives, like `report-uri` and `frame-ancestors`, which are not supported in meta tags.
+
+By default prerendered pages will still use meta tags, but there is now core support for generating headers for prerendered pages as well. Eventually this will be implemented in all adapters, but currently only the `@astrojs/netlify` adapter supports it. To enable this, add the `experimentalStaticHeaders` option to your adapter configuration:
```

Comment:
Should we mention the version of the adapter where `experimentalStaticHeaders` is available? The version is `6.4.0`